LocalSettings.php

```diff
@@ -61,27 +75,35 @@
 # Experimental charset support for MySQL 4.1/5.0.
 $wgDBmysql5 = true;
 
-# File Cache
-#$wgUseFileCache = true; /* default: false */
-#$wgFileCacheDirectory = "/srv/www/htdocs/cache";
-$wgShowIPinHeader = false;
-
-## Shared memory settings
-$wgMemCachedServers = array( 0 => '127.0.0.1:11211' );
-$wgMainCacheType      = CACHE_MEMCACHED;
-$wgMessageCacheType = CACHE_ANYTHING;
-$wgParserCacheType = CACHE_MEMCACHED;
-$configdate = gmdate( 'YmdHis', @filemtime( __FILE__ ) );
-$wgCacheEpoch = max( $wgCacheEpoch, $configdate );
-$wgEnableSidebarCache = true;
-$wgCacheDirectory = "/srv/www/cache";
+## Cache
+if ( $is_production ) {
+	# File Cache
+	#$wgUseFileCache = true; /* default: false */
+	#$wgFileCacheDirectory = "/srv/www/htdocs/cache";
+	$wgShowIPinHeader = false;
+
+	$wgMemCachedServers = array( 0 => '127.0.0.1:11211' );
+	$wgMainCacheType = CACHE_MEMCACHED;
+	$wgMessageCacheType = CACHE_ANYTHING;
+	$wgParserCacheType = CACHE_MEMCACHED;
+	$configdate = gmdate( 'YmdHis', @filemtime( __FILE__ ) );
+	$wgCacheEpoch = max( $wgCacheEpoch, $configdate );
+	$wgEnableSidebarCache = true;
+	$wgCacheDirectory = "/srv/www/cache";
+} else {
+	$wgMainCacheType = CACHE_NONE;
+	$wgCacheDirectory = "$IP/cache";
+	$wgCachePages = false;
+}
 
 ## To enable image uploads, make sure the 'images' directory
 ## is writable, then uncomment this:
 $wgEnableUploads  = true;
-$wgUseImageResize = true;
-$wgUseImageMagick = false;
-#$wgImageMagickConvertCommand = "/usr/bin/convert";
+$wgUseImageMagick = true;
+$wgImageMagickConvertCommand = "/usr/bin/convert";
```

Comment:
What's the reason for enabling ImageMagick? In production, everything works without it ;-)

Reply:
Used to have some uploading failures but now it works well. So I take off ImageMagick now.